File: 1022-sum-of-root-to-leaf-binary-numbers/1022-sum-of-root-to-leaf-binary-numbers.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int dfs(TreeNode t,int sum){
        if(t==null)return 0;
        sum=sum*2 + t.val;
        if(t.left==null && t.right==null)return sum;
        return dfs(t.left,sum)+dfs(t.right,sum);
    }
    public int sumRootToLeaf(TreeNode root) {
        int sum=0;
        return dfs(root,sum);
    }
}

File: 1022-sum-of-root-to-leaf-binary-numbers/1022-sum-of-root-to-leaf-binary-numbers.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int dfs(TreeNode*t,int curr_sum){
        if(!t)return 0;
        curr_sum=curr_sum*2+t->val;
        if(!t->left && !t->right)return curr_sum;
        return dfs(t->left,curr_sum)+dfs(t->right,curr_sum);
    }
    
    int sumRootToLeaf(TreeNode* root) {
        int curr_sum=0;
        return dfs(root,curr_sum);
    }
};